### ISSUE TITLE
Proxy Support

### ISSUE THREAD

**Issue by olegkap** [gregsdennis/Manatee.Trello#251]
I am not sure if its already supported and I didn't find an example but would like to see Proxy Support.
I have an application that runs on internal network where access to the internet is only thru the proxy.

The only alternative I could see is to for your library and expose WebApiClient or HttpClient it uses somehow so I could provide constructor with proxy parameters.
Or another option to add ability to provide your own RestClientProvider and replace DefaultRestClientProvider.

I am using .NET Core 2.1.2 and didn't find any way to specify default proxy so HttpClient can use by default.

Thanks,
Oleg

**Comment by gregsdennis** [comment]
You can definitely [create your own client provider](https://gregsdennis.github.io/Manatee.Trello/usage/supplying-your-own-rest-client.html).  However, it doesn't seem that wrapping the `DefaultRestClientProvider` is an option since the other portions of it are internal.  Maybe that's an oversight on my part.  I have a factory for JSON object creation.  It wouldn't be a departure to add something for `HttpClient`.

Secondarily, wouldn't the proxy setting be controlled by the computer that your app is running on?  Or are you wanting to use different proxy settings from the default settings?

**Comment by olegkap** [comment]
Unfortunately it seems that in .NET Core you cannot use default proxy setting and have to provide your own, I think its due to the fact that .NET Core could be running on different platforms and it could very from platform to platform. Maybe I couldnt find it but in case of GitHub I had to provide my own HttpClient which was configured to use proxy.

I will take a look at creating my own provider then, it seems like the right way to do it, without too many modification and to follow your API.

Thanks,
Oleg

**Comment by olegkap** [comment]
ok, thanks,

I managed to implement it but I found a lot of internal classes which in my opinion shouldn't be.
I simply copy pasted them to finish implementation so at the end I had to implement 5 classes where I had to make a change only to one of them so I think its a little too much but it works.

Also, while you there, think of situation where your client might be used in situation where DI containers are available, such ASP.NET applications since most of your examples assume console or desktop client I assume.

Thanks,
Oleg

**Comment by gregsdennis** [comment]
Yeah, the current implementation considers the case where a wholesale replacement is used, not a modification of what's there.  That's why all the implementation is internal.

**Comment by gregsdennis** [comment]
I haven't forgotten about this.  I'm just not sure what I want to do with it, yet.  I've also been really busy with other projects (and work).

**Comment by gregsdennis** [comment]
@olegkap I'm adding a factory func to the configuration that the default implementation uses to instantiate clients.  You can use this to provide custom initialization, including proxy settings.

See branch **feature/251-client-initialization**.
